internal/interpreter/api.rs

```diff
@@ -148,6 +148,182 @@ impl Value {
             _ => ValueType::Other,
         }
     }
+
+    /// Create a `Value` from a JSON Value
+    #[cfg(feature = "internal-json")]
+    pub fn from_json(
+        t: &i_slint_compiler::langtype::Type,
+        v: &serde_json::Value,
+    ) -> Result<Self, String> {
+        use i_slint_compiler::langtype;
+
+        use smol_str::ToSmolStr;
+
+        fn string_to_color(s: &str) -> Option<i_slint_core::Color> {
+            if !s.starts_with('#') || s.len() != 9 {
+                return None;
+            }
+
+            let c = u32::from_str_radix(&s[1..s.len()], 16).ok()?;
```

Comment:
```suggestion
            let c = u32::from_str_radix(&s[1..], 16).ok()?;
```

But this function is only parsing with 8 number and i'm not sure this is portable (eg, big endian)

There is also i_slint_compiler::literals::parse_color_literal that you could use.

Reply:
OK, I'll use parse_color_literal. I was avoiding that to have a stricter parser for machine-generated code, but thinking about this again, that is not a good reason considering I expect at least some people to work with the JSON.